src/lib/mina_graphql/mina_graphql.ml

```diff
@@ -2870,41 +2876,85 @@ module Queries = struct
         List.map best_chain ~f:(block_of_breadcrumb coda) )
 
   let block =
-    result_field "block"
+    result_field2 "block"
       ~doc:
-        "Retrieve a block with the given state hash, if contained in the \
-         transition frontier."
+        "Retrieve a block with the given state hash or height, if contained \
+         in the transition frontier."
       ~typ:(non_null Types.block)
       ~args:
         Arg.
           [ arg "stateHash" ~doc:"The state hash of the desired block"
-              ~typ:string ]
+              ~typ:string
+          ; arg "height" ~doc:"The height of the desired block" ~typ:int ]
       ~resolve:
-        (fun {ctx= coda; _} () (state_hash_base58_opt : string option) ->
-        let open Result.Let_syntax in
-        let%bind state_hash_base58 =
-          state_hash_base58_opt
-          |> Result.of_option ~error:"Must provide a state hash"
-        in
-        let transition_frontier_pipe = Mina_lib.transition_frontier coda in
-        let%bind transition_frontier =
-          Pipe_lib.Broadcast_pipe.Reader.peek transition_frontier_pipe
-          |> Result.of_option ~error:"Could not obtain transition frontier"
-        in
-        let%bind state_hash =
-          State_hash.of_base58_check state_hash_base58
-          |> Result.map_error ~f:Error.to_string_hum
-        in
-        let%map breadcrumb =
-          Transition_frontier.find transition_frontier state_hash
-          |> Result.of_option
-               ~error:
-                 (sprintf
-                    "Breadcrumb for state hash %s not found in transition \
-                     frontier"
-                    state_hash_base58)
-        in
-        block_of_breadcrumb coda breadcrumb )
+        (fun {ctx= coda; _} () (state_hash_base58_opt : string option)
+             (height_opt : int option) ->
+        let have_state_hash = Option.is_some state_hash_base58_opt in
+        let have_height = Option.is_some height_opt in
```

Comment:
nit: I think this function would be cleaner using a pattern match on `(state_hash_opt, height_opt)`, then you won't need the value_exn later.

As written, it looks correct to me. But I'm worried that future edits to this function could introduce crashes accidentally when we're relying on the value_exns.

But I won't block the PR on it -- if you do want to change it, let me know and I'll look at it again